### PR TITLE
[#244] Provide clear message on OpenID connection error

### PR DESCRIPTION
**Pull request by MartinFlores751**
```
$ irods_http_api config.json 
Validating configuration file ...
No JSON schema file provided. Using default.
Configuration passed validation!
[2024-11-05 18:54:10.724] [P:15090] [info] [T:15090] Initializing server.
[2024-11-05 18:54:10.724] [P:15090] [trace] [T:15090] Verifying OIDC endpoint configuration
[2024-11-05 18:54:10.724] [P:15090] [trace] [T:15090] Unable to establish connection to OpenID Provider. Reason: [connect: Connection refused [system:111 at /opt/irods-externals/boost-libcxx1.81.0-1/include/boost/asio/detail/reactive_socket_service.hpp:587:33 in function 'connect']]
[2024-11-05 18:54:10.724] [P:15090] [error] [T:15090] Invalid OIDC configuration, server not starting.
```

### CODE REVIEW

**Review by korydraughn — APPROVED** [review]
Looks good to me.

Let's wait for one more approval before pounding.

File: core/src/main.cpp
Comment (trel) [inline core/src/main.cpp:823]: let's go `debug` or even `info` here...    feels useful to an admin at all times to see this message.

Comment (korydraughn) [reply core/src/main.cpp:823]: Agreed. Let's do it.

Comment (MartinFlores751) [reply core/src/main.cpp]: Done.

**Comment by korydraughn** [comment]
Squash it.

**Comment by MartinFlores751** [comment]
Squashed.

**Comment by korydraughn** [comment]
There's one commit without an issue number in there.

**Comment by trel** [comment]
i only see one commit - looks good?

**Comment by korydraughn** [comment]
I see 3 additional commits? Perhaps my browser is showing old information.

**Comment by MartinFlores751** [comment]
The 3 additional commits are probably the accidental references to this commit I made. The commits belong to a separate PR.

**Review by korydraughn — APPROVED** [review]
Pound it.

**Comment by MartinFlores751** [comment]
Pounded.

**Pull request merged**
